docs/advanced_topics/migrating.rst

```diff
@@ -0,0 +1,114 @@
+
+When making changes to a Wagtail page's field schema we sometimes have to change the data for those fields on any pages that exist already in the database. See the  `Django data migrations documentation <https://docs.djangoproject.com/en/stable/topics/migrations/#data-migrations>`_ for a detailed description of how data migrations work. Data migrations are straight-forward for :class:`~wagtail.core.models.Page` obejcts themselves, but pages also have historical revisions and potentially draft revisions and each of this :class:`~wagtail.core.models.PageRevision` objects also needs to be migration.
+
+Modifying page revisions
+------------------------
+
+:class:`~wagtail.core.models.PageRevision` objects store the :class:`~wagtail.core.models.Page` object at the time the revision was created as a JSON string in :class:`~wagtail.core.models.PageRevision.content_json`. Any modification of the revision has to load this JSON, modify it, and then dump it back to a string to store it. Given a particular ``revision`` object:
+
+.. code-block:: python
+
+    # Load the revision content JSON
+    revision_content = json.loads(revision.content_json)
+    # Migrate the revision content
+    revision_content['name'] =  ' '.join(
+        revision_content['first_name'], 
+        revision_content['last_name']
+    )
+    # Dump the revision back to JSON and store it
+    revision.content_json = json.dumps(revision_content)
+
+Creating the data migration
+---------------------------
+
+Knowing that we need to modify both the page objects and the page revision objects, we have to get both models from the `Django app registry <https://docs.djangoproject.com/en/2.0/ref/applications/>`_. From there we can loop over all the page objects and modify them as necessary, and then loop over each of the page objects' revision and modify them as described above.
+
+.. code-block:: python
+
+    from django.db import migrations
+
+
+    def combine_names(apps, schema_editor):
+        # Get the page and revision models
+        page_model = apps.get_model('myapp', 'PageModel')
+        revision_model = apps.get_model('wagtailcore.PageRevision')
+
+        for page in page_model.objects.all():
+            # Migrate the page object
+            page.name = ' '.join(page.first_name, page.last_name)
+
+            revisions = revision_model.objects.filter(page=page)
+            for revision in revisions:
+                # Load the revision content JSON
+                revision_content = json.loads(revision.content_json)
+                # Migrate the revision content
+                revision_content['name'] =  ' '.join(
+                    revision_content['first_name'], 
+                    revision_content['last_name']
+                )
+                # Dump the revision back to JSON and store it
+                revision.content_json = json.dumps(revision_content)
+
+
+    class Migration(migrations.Migration):
+        dependencies = []
```

Comment:
Defining `dependencies` here as an empty list might be confusing for people who aren't so familiar with migration files. Would you consider either excluding this or copying over `('yourappname', '0001_initial'),` from [the Django example](https://docs.djangoproject.com/en/2.0/topics/migrations/#data-migrations)?